### PR TITLE
GITHUB-APD-137: Integrate cards in the panel

### PR DESCRIPTION
**Pull request by christophe-chausseray**
**Description (for Contributor and Core Developer)**

Ticket : https://github.com/akeneo/actionable-product-data/issues/137

Kapture 2020-06-05 at 14 14 00

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added unit tests                      | OK
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import {CampaignFetcherImplementation} from 'akeneocommunicationchannel/fetcher/campaign';
+import {PanelDataProvider} from 'akeneocommunicationchannel/components/panel';
+
+const dataProvider: PanelDataProvider = {
+  cardFetcher: CardFetcherImplementation,
+  campaignFetcher: CampaignFetcherImplementation
+};
+
+const Index = () => (
+  <AkeneoThemeProvider>
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/index.tsx:14]: Maybe you could use the `legacy-bridge` provider and benefit from the hooks to access legacy stuff, such as the router and translator, like here https://github.com/akeneo/pim-community-dev/blob/master/src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/quantified-associations/QuantifiedAssociationsTab.tsx#L7

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/card/Link.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import styled from 'styled-components';
+import {AkeneoThemedProps} from '@akeneo-pim-community/shared';
+
+const __ = require('oro/translator');
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/card/Link.tsx:5]: And then if you use the `legacy-bridge` provider you could
```suggestion
import {useTranslate} from '@akeneo-pim-community/legacy-bridge';
```
And use it as `const translate = useTranslate();`

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+const Title = styled.div`
+  color: ${({theme}: AkeneoThemedProps) => theme.color.white};
+  font-size: 22px;
+  height: 27px;
+  float: left;
+  margin: 20px 0 0 10px;
+  letter-spacing: 0.92px;
+`;
+
+const CloseButton = styled.div`
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Header.tsx:30]: Don't know if it suits your needs but there also are some "JSXified" icons in the `shared` workspace that you could reuse
https://github.com/akeneo/pim-community-dev/blob/master/src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/icons/CloseIcon.tsx

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/card/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import Link from 'akeneocommunicationchannel/components/panel/card/Link';
+import {AkeneoThemedProps} from '@akeneo-pim-community/shared';
+import {ellipsisText} from '../../../tools/ellipsis-text';
+
+const Container = styled.li`
+  margin: 20px;
+  padding-bottom: 20px;
+
+  &:not(:last-child) {
+    border-bottom: 1px solid #c7cbd4;
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/card/Card.tsx:14]: Not a theme color?

Comment (christophe-chausseray) [reply src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/card/Card.tsx:14]: Yes some colors in the design aren't in the theme. I have to see with the UX designer they have it in their DSM to add the same name in our akeneo theme.

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Header.tsx (hunk 3 above)
Comment (christophe-chausseray) [reply src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Header.tsx:31]: Thanks for the tips. I updated it :)

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      <IconContainer><MegaphoneIcon color={akeneoTheme.color.purple100} size={20} /></IconContainer>
+      <Title>{title}</Title>
+      <CloseButton  onClick={onClickCloseButton}>
+        <CloseIcon color={akeneoTheme.color.purple100} title={__('pim_common.close')} size={15} />
+      </CloseButton>
+    </Container>
+  );
+};
+
+export = HeaderPanelComponent;
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Header.tsx:59]:
```suggestion
export {HeaderPanelComponent};
```

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Panel.tsx [hunk 6]
```diff
@@ -1,54 +1,47 @@
...
-import {CardFetcherImplementation, CardFetcher} from 'akeneocommunicationchannel/fetcher/card';
+import {CardFetcher} from 'akeneocommunicationchannel/fetcher/card';
+import {useCampaign} from 'akeneocommunicationchannel/hooks/useCampaign';
+import {CampaignFetcher} from 'akeneocommunicationchannel/fetcher/campaign';
+import HeaderPanel from 'akeneocommunicationchannel/components/panel/Header';
+import {CardComponent} from 'akeneocommunicationchannel/components/panel/card';
+import {Card} from 'akeneocommunicationchannel/models/card';
 
 const mediator = require('oro/mediator');
 const __ = require('oro/translator');
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Panel.tsx:12]:
```suggestion
import {useTranslate} from '@akeneo-pim-community/legacy-bridge';
```
😉

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/index.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import {CampaignFetcherImplementation} from 'akeneocommunicationchannel/fetcher/campaign';
+import {PanelDataProvider} from 'akeneocommunicationchannel/components/panel';
+
+const dataProvider: PanelDataProvider = {
+  cardFetcher: CardFetcherImplementation,
+  campaignFetcher: CampaignFetcherImplementation
+};
+
+const Index = () => (
+  <DependenciesProvider>
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/index.tsx:15]: 👍

**Comment by ahocquard** [comment]
Nota big fan of the naming "campaign". Personally, I use the name "announcements".

File: src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Panel.tsx (hunk 6 above)
Comment (christophe-chausseray) [reply src/Akeneo/Platform/Bundle/CommunicationChannelBundle/Resources/public/components/panel/Panel.tsx:12]: I update it in the next PR :)

**Pull request merged**